src/couch_prometheus/src/couch_prometheus_sup.erl

```diff
@@ -0,0 +1,39 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_prometheus_sup).
+
+-behaviour(supervisor).
+
+-export([
+    start_link/0,
+    init/1
+]).
+
+-define(CHILD(I, Type), {I, {I, start_link, []}, permanent, 5000, Type, [I]}).
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+    {ok, {
+        {one_for_one, 5, 10}, [
```

Comment:
how were these numbers chosen?

Reply:
if the couch_prometheus_server process crashing should the couch_prometheus_http be restarted? (i.e, rest_for_one).

Reply:
The values were copied over directly from couch_stats without much consideration. I think I can make this either configurable or have higher numbers? 

You'e right in that doesn't make sense for  `couch_prometheus_http` to exist without `couch_prometheus_server`, but does restarting really enforce that? I guess if `couch_prometheus_server` keeps crashing till it doesn't crash anymore, then we don't start `couch_prometheus_http`.